kroxylicious-integration-tests/src/test/java/io/kroxylicious/proxy/schema/validation/JsonSchemaValidationIT.java

```diff
@@ -0,0 +1,189 @@
+              "$schema": "http://json-schema.org/draft-07/schema#",
+              "title": "Person",
+              "type": "object",
+              "properties": {
+                "firstName": {
+                  "type": "string",
+                  "description": "The person's first name."
+                },
+                "lastName": {
+                  "type": "string",
+                  "description": "The person's last name."
+                },
+                "age": {
+                  "description": "Age in years which must be equal to or greater than zero.",
+                  "type": "integer",
+                  "minimum": 0
+                }
+              }
+            }
+            """;
+
+    private static final String JSON_MESSAGE = "{\"firstName\":\"json1\",\"lastName\":\"json2\"}";
+    private static final String INVALID_AGE_MESSAGE = "{\"firstName\":\"json1\",\"lastName\":\"json2\",\"age\":-3}";
+    private static final String TOPIC_1 = "my-test-topic";
+    private static final String TOPIC_2 = "my-test-topic-2";
+
+    private static final String APICURIO_REGISTRY_HOST = "http://localhost";
+    private static final Integer APICURIO_REGISTRY_PORT = 8081;
+    private static final String APICURIO_REGISTRY_URL = APICURIO_REGISTRY_HOST + ":" + APICURIO_REGISTRY_PORT;
+
+    private static GenericContainer registryContainer;
+
+    @BeforeAll
+    public static void init() throws IOException {
+        // An Apicurio Registry instance is required for this test to work, so we start one using a Generic Container
+        DockerImageName dockerImageName = DockerImageName.parse("quay.io/apicurio/apicurio-registry-mem")
+                .withTag("2.5.11.Final");
+
+        Consumer<CreateContainerCmd> cmd = e -> e.withPortBindings(
+                new PortBinding(Ports.Binding.bindPort(APICURIO_REGISTRY_PORT), new ExposedPort(APICURIO_REGISTRY_PORT)));
+
+        registryContainer = new GenericContainer<>(dockerImageName)
+                .withEnv(Map.of(
+                        "QUARKUS_HTTP_PORT", String.valueOf(APICURIO_REGISTRY_PORT),
+                        "REGISTRY_APIS_V2_DATE_FORMAT", "yyyy-MM-dd'T'HH:mm:ss'Z'"))
+                .withExposedPorts(APICURIO_REGISTRY_PORT)
+                .withCreateContainerCmdModifier(cmd);
+
+        registryContainer.start();
+        registryContainer.waitingFor(Wait.forLogMessage(".*Installed features:*", 1));
+
+        // Preparation: In this test class, a schema already registered in Apicurio Registry with globalId one is expected, so we register it upfront.
+        try (var client = RegistryClientFactory.create(APICURIO_REGISTRY_URL)) {
+            client.createArtifact(null, UUID.randomUUID().toString(), IoUtil.toStream(JSON_SCHEMA_TOPIC_1));
+        }
+    }
+
+    @Test
+    void testValidJsonProduceAccepted(KafkaCluster cluster, Admin admin) throws Exception {
+        assertThat(cluster.getNumOfBrokers()).isOne();
```

Comment:
cluster injection is guaranteed to give you one broker, so the assertion seems excessive.  
The test framework has the capability to give you a test topic.  Just add a parameter of type io.kroxylicious.testing.kafka.junit5ext.Topic.